### PR TITLE
Cluster improvements

### PR DESCRIPTION
**Pull request by mkurek**
- added table with master information to cluster
- new cluster fields: status, hostname
- better handling of cluster base objects in API

### CODE REVIEW

File: src/ralph/api/tests/_base.py [hunk 1]
```diff
@@ -69,3 +69,10 @@ def setUpClass(cls):
     def setUp(self):
         super().setUp()
         self.client.force_authenticate(self.superuser)
+
+    @classmethod
+    def get_full_url(self, url):
+        """
+        Convert URL to fully qualified URL (with protocol and domain).
```
Comment (mkurek) [inline src/ralph/api/tests/_base.py]: add note about https://github.com/django/django/blob/master/django/test/client.py#L283

File: src/ralph/data_center/admin.py [hunk 2]
```diff
@@ -84,7 +99,35 @@ class ClusterBaseObjectInline(RalphTabularInline):
         extra = 1
         verbose_name = _('Base Object')
 
-    inlines = [ClusterBaseObjectInline]
+    inlines = [ClusterBaseObjectInline]  # TODO: add ClusterNetworkInline
```
Comment (szok) [inline src/ralph/data_center/admin.py]: Add ClusterNetworkInline  to inlines :)

File: src/ralph/data_center/models/virtual.py [hunk 3]
```diff
@@ -26,18 +30,73 @@ def __str__(self):
...
+
+class ClusterStatus(Choices):
+    _ = Choices.Choice
 
+    in_use = _('in use')
+    for_deploy = _('for deploy')
 
-class Cluster(BaseObject, NamedMixin, models.Model):
 
+class Cluster(BaseObject, models.Model):
```
Comment (szok) [inline src/ralph/data_center/models/virtual.py:49]: Why not use NamedMixin?

File: src/ralph/lib/table/table.py [hunk 4]
```diff
@@ -24,10 +24,10 @@ class Table(object):
     Generating contents for table based on predefined columns and queryset.
 
     Example:
-        >>> table = Table(queryset, ['id', 'name'])
+        >>> table = Table(queryset, ['id', ('name', 'My field name')])
```
Comment (ghost) [inline src/ralph/lib/table/table.py:26]: Could you mention about `transpose` option?
What's the meaning of tuple `('name', 'My field name')`?

File: src/ralph/data_center/models/virtual.py (hunk 3 above)
Comment (mkurek) [inline src/ralph/data_center/models/virtual.py:49]: because in our case this field could be blank/null

File: src/ralph/lib/table/tests.py [hunk 5]
```diff
@@ -13,17 +13,34 @@ def setUpClass(cls):
...
+        self.assertEqual(len(table1), len(table2))
+        for row1, row2 in zip(table1, table2):
+            self.assertSequenceEqual(row1, row2)
+
     def test_queryset_render(self):
         """Test get_perm_key function."""
         table = Table(Foo.objects.all(), ['id', 'bar'])
         result = [
-            ['ID', 'bar'],
+            [{'value': 'ID'}, {'value': 'bar'}],
```
Comment (ghost) [inline src/ralph/lib/table/tests.py:28]: Why ID here is capsed?

File: src/ralph/data_center/models/virtual.py (hunk 3 above)
Comment (szok) [inline src/ralph/data_center/models/virtual.py:49]: ok :)

File: src/ralph/lib/table/table.py (hunk 4 above)
Comment (xor-xor) [inline src/ralph/lib/table/table.py:26]: +1 re: meaning of this tuple.

Comment (mkurek) [inline src/ralph/lib/table/table.py:26]: i've added some description for **init** params. check it out :)

File: src/ralph/lib/table/tests.py (hunk 5 above)
Comment (mkurek) [inline src/ralph/lib/table/tests.py:28]: this is verbose name of this field. https://github.com/django/django/blob/master/django/db/models/options.py#L254

File: src/ralph/lib/table/table.py (hunk 4 above)
Comment (xor-xor) [inline src/ralph/lib/table/table.py:26]: That description is fine, but I'd also add here a remark: "see **init**'s docstring for additional info" (or something like that).

Comment (mkurek) [inline src/ralph/lib/table/table.py:26]: done

**Comment by szok** [comment]
:ok:

**Comment by ghost** [comment]
:ok:

**Pull request merged**